src/Compilers/CSharp/Test/Syntax/Parsing/StackAllocInitializerTests.cs

```diff
@@ -0,0 +1,112 @@
+            UsingExpression("stackalloc int[] { 42 }", options: TestOptions.Regular7,
+                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion7, "stackalloc").WithArguments("stackalloc initializer", "7.3").WithLocation(1, 1));
+            N(SyntaxKind.StackAllocArrayCreationExpression);
+            {
+                N(SyntaxKind.StackAllocKeyword);
+                N(SyntaxKind.ArrayType);
+                {
+                    N(SyntaxKind.PredefinedType);
+                    {
+                        N(SyntaxKind.IntKeyword);
+                    }
+                    N(SyntaxKind.ArrayRankSpecifier);
+                    {
+                        N(SyntaxKind.OpenBracketToken);
+                        N(SyntaxKind.OmittedArraySizeExpression);
+                        {
+                            N(SyntaxKind.OmittedArraySizeExpressionToken);
+                        }
+                        N(SyntaxKind.CloseBracketToken);
+                    }
+                }
+                N(SyntaxKind.ArrayInitializerExpression);
+                {
+                    N(SyntaxKind.OpenBraceToken);
+                    N(SyntaxKind.NumericLiteralExpression);
+                    {
+                        N(SyntaxKind.NumericLiteralToken, "42");
+                    }
+                    N(SyntaxKind.CloseBraceToken);
+                }
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void StackAllocInitializer_02()
+        {
+            UsingExpression("stackalloc int[1] { 42 }", options: TestOptions.Regular7,
+                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion7, "stackalloc").WithArguments("stackalloc initializer", "7.3").WithLocation(1, 1));
+            N(SyntaxKind.StackAllocArrayCreationExpression);
+            {
+                N(SyntaxKind.StackAllocKeyword);
+                N(SyntaxKind.ArrayType);
+                {
+                    N(SyntaxKind.PredefinedType);
+                    {
+                        N(SyntaxKind.IntKeyword);
+                    }
+                    N(SyntaxKind.ArrayRankSpecifier);
+                    {
+                        N(SyntaxKind.OpenBracketToken);
+                        N(SyntaxKind.NumericLiteralExpression);
+                        {
+                            N(SyntaxKind.NumericLiteralToken, "1");
+                        }
+                        N(SyntaxKind.CloseBracketToken);
+                    }
+                }
+                N(SyntaxKind.ArrayInitializerExpression);
+                {
+                    N(SyntaxKind.OpenBraceToken);
+                    N(SyntaxKind.NumericLiteralExpression);
+                    {
+                        N(SyntaxKind.NumericLiteralToken, "42");
+                    }
+                    N(SyntaxKind.CloseBraceToken);
+                }
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void StackAllocInitializer_03()
```

Comment:
Could you also test `stackalloc [1] { 42 }`? 
My feeling is that the parser should be able to handle and recover (then the binding logic would reject it). Tagging @gafter for advice. 
Maybe a way to do that would be to extend the `StackAllocArrayCreationExpressionSyntax` syntax node to have an optional type (and rename it to `TypeOpt`), rather than introducing a new syntax node.

Reply:
in a `StackAllocArrayCreationExpressionSyntax` braces are a part of the type (which is optional now), but `ImplicitStackAllocArrayCreationExpressionSyntax` has braces of its own. so I guess they cannot be combined unless you want a couple of mutually exclusive nodes.

Reply:
>`stackalloc [1] { 42 }`

this won't give meaningful diagnostics for both `stackalloc` and `new`, only syntax errors.